### PR TITLE
[Combinatorics] Adding support for BigNumber in StirlingS2 and BellNumbers.

### PR DESCRIPTION
**Pull request by devanp92**

### CODE REVIEW

**Comment by devanp92** [comment]
This should have been split into 2 separate PR's. One for fixing the StirlingS2 and BellNumbers (for BigNumber support) and another for the Composition function. I can reset the commit if needed.

**Comment by josdejong** [comment]
Thanks! It's no problem to add more commits on an open PR in general, and it's very related :)

File: lib/function/combinatorics/composition.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      // not all numbers can be converted to BigNumber
+      k = BigNumber.convert(k);
+
+      if (!(k instanceof BigNumber) || !isPositiveInteger(n) || !isPositiveInteger(k)) {
+        throw new TypeError('Positive integer value expected in function composition');
+      }
+      if (k.gt(n)) {
+        throw new TypeError('k must be less than n in function composition');
+      }
+      return math.combinations(n-1,k-1);
```
Comment (josdejong) [inline lib/function/combinatorics/composition.js]: What will happen here is that both BigNumbers n and k become stringified, then converted to a number, and then gets 1 subtracted.

To do real BigNumber computation, it should be like: 
```js
return math.combinations(n.minus(1), k.minus(1));
```

Maybe you can add some unit tests which will fail (overflow) with regular numbers and can only succeed using BigNumbers?

**Comment by devanp92** [comment]
This has been fixed.

**Comment by josdejong** [comment]
Thanks, I will check it out asap

**Comment by josdejong** [comment]
Thanks, looks good with using the math functions, so you automatically handle both numbers and bignumbers. 

I saw an issue in each of the functions, you use some comparisons to check whether `n` is a positive integer:

```js
if (isNumber(n) || n instanceof BigNumber) {

  if (!isInteger(n) || n < 0) {
    throw new TypeError('Positive integer value expected in function bellNumbers');
  }
```

The problem here is that the function `isInteger` and the operator `<` only work on numbers. So this will fail for large BigNumbers for which the number representation becomes infinity. Would be great if you could improve on this.

**Comment by devanp92** [comment]
Do you think what I wrote in [362](https://github.com/josdejong/mathjs/pull/362) would work for these functions?

**Comment by devanp92** [comment]
Do you think what I wrote in [362](https://github.com/josdejong/mathjs/pull/362) would work for these functions?

**Comment by josdejong** [comment]
Yes I think so, what you posted there is more or less the idea of this helper function `isPositiveInteger`.

**Comment by devanp92** [comment]
This has been fixed in [362](https://github.com/josdejong/mathjs/pull/362).

**Comment by josdejong** [comment]
@devanp92 do you mean this PR can be closed?

**Comment by devanp92** [comment]
Yes!

**Comment by josdejong** [comment]
oki :)
